src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs

```diff
@@ -55,12 +55,32 @@ public static string GetNuGetLockFilePath(string baseDirectory, string projectNa
             return Path.Combine(baseDirectory, PackagesLockFileFormat.LockFileName);
         }
 
+        /// <summary>
+        /// The lock file will get invalidated if one or more of the below are true
+        ///     1. The target frameworks list of the current project was updated.
+        ///     2. The runtime list of the current project waw updated.
+        ///     3. The packages of the current project were updated.
+        ///     4. The packages of the dependent projects were updated.
+        ///     5. The framework list of the dependent projects were updated with frameworks incompatible with the main project framework.
+        /// </summary>
+        /// <param name="dgSpec">The <see cref="DependencyGraphSpec"/> for the new project defintion.</param>
+        /// <param name="nuGetLockFile">The current <see cref="PackagesLockFile"/>.</param>
+        /// <returns></returns>
         public static bool IsLockFileStillValid(DependencyGraphSpec dgSpec, PackagesLockFile nuGetLockFile)
         {
             var uniqueName = dgSpec.Restore.First();
             var project = dgSpec.GetProjectSpec(uniqueName);
 
             // Validate all the direct dependencies
+            var lockFileFrameworks = nuGetLockFile.Targets
+                .Where(t => t.TargetFramework != null)
+                .Select(t => t.TargetFramework)
+                .Distinct();
```

Comment:
Comment ported from https://github.com/NuGet/NuGet.Client/pull/2920#discussion_r303684691
@nkolev92 said: 
I would remove the distinct here.
I think it'd only hide potential bugs. Those values should be unique.

@cristinamanum answer:
A project set as:  <PropertyGroup>
    <TargetFrameworks>netcoreapp2.2</TargetFrameworks>
    <RuntimeIdentifiers>win-x86;osx-x64</RuntimeIdentifiers>
  </PropertyGroup>

Will get three dependencies in the lock file all off them with the same framework.

Reply:
I understand. Makes sense.